src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -38,6 +39,13 @@ class graph_impl
     , handle(ghdl)
   {}
 
+  graph_impl(xrt::hw_context hwctx, const std::string& name, graph::access_mode am)
+      : hw_ctx(std::move(hwctx))
+  {
+    device = hwctx.get_device().get_handle();
+    handle = device->open_graph(hwctx, name.c_str(), am);
+  }
```

Comment:
This is now a bug since you can't use `hwctx` after it was moved into `hw_ctx`.

Use initializer list and initialize in order of declaration in class.

```suggestion
  graph_impl(xrt::hw_context hwctx, const std::string& name, graph::access_mode am)
      : device{hwctx.get_device().get_handle()}
      , hw_ctx{std::move(hwctx)}
      , handle{device->open_graph(hw_ctx, name.c_str(), am)}
  {}
```